### PR TITLE
Added an Upload Button for Files

### PR DESCRIPTION
**Pull request by Denton-L**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -327,6 +331,30 @@ var run = function() {
 		})
 	});
 
+	$('.inputFile').on('change', function() {
+		var fileList = $('input.inputFile')[0].files;
+		for (var i = 0; i < fileList.length; i++) {
+			SOL_CACHE_FILE = fileKey(fileList[i].name);
```
Comment (chriseth) [inline src/app.js]: I think you have to adjust that following the latest refactoring.

File: src/app.js [hunk 2]
```diff
@@ -327,6 +331,30 @@ var run = function() {
...
+			(function(name, file) {
+				var fileReader = new FileReader();
+				fileReader.onload = function(e) {
+					if ((!window.localStorage[name]) ||
+							(window.localStorage[name] && confirm("The file " +fileNameFromKey(name) +" already exists! Would you like to overwrite it?"))) {
```
Comment (chriseth) [inline src/app.js]: The part `window.localStorage[name] &&` should not be necessary.

**Comment by Denton-L** [comment]
Alright, I have made the changes.

File: src/app.js [hunk 3]
```diff
@@ -327,6 +331,25 @@ var run = function() {
...
+				fileReader.onload = function(e) {
+					SOL_CACHE_FILE = fileKey(file.name);
+					window.localStorage[SOL_CACHE_FILE] = e.target.result;
+					updateFiles();
+				}
+				fileReader.readAsText(file);
+			})(fileList[i]);
+		}
+
+		$filesEl.animate({left: Math.max( (0 - activeFilePos() + (FILE_SCROLL_DELTA/2)), 0)+ "px"}, "slow", function(){
```
Comment (axic) [inline src/app.js]: This (and all the above `function()`s) don't conform to the new lint. Please run `semistandard app.js` for a list of errors (we haven't addressed the indentation errors it spits out, yet)

**Comment by axic** [comment]
Can you squash the relevant changes into fewer commits? See `git rebase -i HEAD~11`.

**Comment by axic** [comment]
Can you also rebase this against master (`git rebase master` on the `upload-button` branch) ? Github seems to make the diff look weird.

**Comment by Denton-L** [comment]
I squashed it all into a single commit. Don't pull yet, though, I'm going to move everything into `editor.js`.

File: src/app.js [hunk 4]
```diff
@@ -157,6 +161,30 @@ var run = function () {
...
+						window.localStorage[name] = e.target.result;
+						sessions[name] = null;
+						updateFiles();
+					}
+				}
+				fileReader.readAsText(file);
+			})(SOL_CACHE_FILE, fileList[i]);
+		}
+
+		$filesEl.animate({left: Math.max( (0 - activeFilePos() + (FILE_SCROLL_DELTA/2)), 0)+ "px"}, "slow", function(){
```
Comment (axic) [inline src/app.js]: Please check how the spacing is in the rest of the file. Should be something similar to:
`+		$filesEl.animate({ left: Math.max((0 - activeFilePos() + (FILE_SCROLL_DELTA / 2)), 0) + "px" }, "slow", function () {`

And all quotation marks should be single quotes (`'`).

**Comment by Denton-L** [comment]
Hopefully, this is all formatted correctly.

**Pull request merged**